provider/common/errors.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import "github.com/juju/errors"
+
+// ZoneIndependentError wraps the given error such that it
+// satisfies environs.IsAvailabilityZoneIndependent.
+func ZoneIndependentError(err error) error {
+	if err == nil {
+		return nil
+	}
+	wrapped := errors.Wrap(err, zoneIndependentError{err})
+	wrapped.(*errors.Err).SetLocation(1)
```

Comment:
This doesn't seem safe (fmt.Errorf() won't return this type, etc)
Can we do:
if traceErr, ok := wrapped.(*errors.Err); ok {
  traceErro.SetLocation(1)
}
That way we don't blind cast and get a panic()

Reply:
as mentioned on IRC: this is operating on the result of errors.Wrap, which is always an \*errors.Err. there's a test, so if that changes then we'll catch it